pennylane_lightning/lightning_qubit.py

```diff
@@ -263,22 +273,35 @@ def reset(self):
     @property
     def state(self):
         # Flattening the state.
-        shape = (1 << self.num_wires,)
+        dim = 1 << self.num_wires
+        batch_size = self._get_batch_size(self._pre_rotated_state, (2,) * self.num_wires, dim)
+        # Do not flatten the state completely but leave the broadcasting dimension if there is one
+        shape = (batch_size, dim) if batch_size is not None else (dim,)
         return self._reshape(self._pre_rotated_state, shape)
 
     def _apply_state_vector(self, state, device_wires):
         """Initialize the internal state vector in a specified state.
+
         Args:
             state (array[complex]): normalized input state of length ``2**len(wires)``
-                or broadcasted state of shape ``(batch_size, 2**len(wires))``
```

Comment:
I'm slightly confused here:
1. Why was this line here before?
2. Why is it removed?

Reply:
This line was here because of a previous attempt at broadcasting. I'm removing it now because in a first approach we are not supporting initializing a state vector in a broadcasted shape.

Reply:
Thanks for clarifying, makes sense :)